src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -79,6 +79,15 @@ Copyright (c) .NET Foundation. All rights reserved.
     <PredefinedCulturesOnly Condition="'$(PredefinedCulturesOnly)' == '' and '$(InvariantGlobalization)' == 'true'">true</PredefinedCulturesOnly>
   </PropertyGroup>
 
+  <!-- Set the IsRidAgnostic property if this project should NOT accept global RuntimeIdentifier and SelfContained
+       property values from referencing projects. -->
+  <PropertyGroup Condition="'$(IsRidAgnostic)' == ''">
+    <IsRidAgnostic Condition="('$(_IsExecutable)' == 'true' And '$(IsTestProject)' != 'true') Or
```

Comment:
The applies to anything that has an executable output type? So both `Exe` and `WinExe`. I noticed msbuild defines a subset of what it seems the compiler supports according to https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/output#targettype

Reply:
I think `Exe` and `WinExe` are the only "executable" values supported by projects using the .NET SDK, and the others are outdated.